cmd/antrea-agent/agent.go

```diff
@@ -87,11 +87,27 @@ func run(o *Options) error {
 	// Create Antrea Clientset for the given config.
 	antreaClientProvider := agent.NewAntreaClientProvider(o.config.AntreaClientConnection, k8sClient)
 
+	agentFeatures := o.getAgentFeatures()
 	// Register Antrea Agent metrics if EnablePrometheusMetrics is set
-	if *o.config.EnablePrometheusMetrics {
+	if agentFeatures.PrometheusMetricsEnabled {
```

Comment:
I don't quite get the point of making another struct for all features based on the existing `DefaultFeatureGate` and options. Even if including all feature state in one struct is required in certain scenarios, `DefaultFeatureGate` is already a well encapsulated interface for querying each feature. But I think the point of previous code that injects individual feature states to modules that need them is meaningful and more OOP, even though the argument list is long: It's a statement that what dependencies these module depend on. Wrapping all configs and feature states in a common struct breaks it. We won't know that dependencies from the initializer and it will just look like every module is related to all features and can be impacted by all configurations.
It doesn't look right to include feature states in `NetworkConfig` to reduce number of arguments either.
I would suggest only to move few networkConfig related flags to `NetworkConfig` as @jianjuns originnally suggested in this PR and leaves the feature gates as is.

Reply:
@antoninbas @jianjuns Could you share your points? Do you think it is needed to save the flags of Antrea features with a specific struct in NetworkConfig?

Reply:
I did not look into all cases of features references. But maybe in this PR let us handle v4 and v6 first. A single struct for all features only makes sense when many components all look up many features.